src/typechecker.ts

```
@@ -0,0 +1,1951 @@
+    } break;
+
+    case AstNodeKind.Expr: {
+      if (!parsed_node.item) return Result.Ok(T.void);
+      const result = get_type(ctx, parsed_node.item);
+      if (!result.ok) return Result.Err(result.error);
+      typed_node = result.value;
+    } break;
+
+    default: {
+      throw new Error(`Unhandled parse node kind attempting to get type: ${parsed_node.kind}`);
+    };
```

Biome (2.1.2)

[error] This code is unreachable



(lint/correctness/noUnreachable)